src/charm.py

```diff
@@ -100,6 +101,17 @@ class VaultOperatorCharm(CharmBase):
     def __init__(self, *args):
         super().__init__(*args)
         self.machine = Machine()
+        self._cos_agent = COSAgentProvider(
+            self,
+            scrape_configs=[
+                {
+                    "scheme": "https",
+                    "tls_config": {"insecure_skip_verify": True},
```

Comment:
Are there any security implications to this?

Reply:
This should be the same as running `curl -k $VAULT_ADDR/v1/sys/metrics?format=prometheus`
We used the same config in the [vault-k8s ](https://github.com/canonical/vault-k8s-operator/blob/main/src/charm.py#L232)charm, and there some other charms that do the same like [postgresql](https://github.com/canonical/postgresql-operator/blob/main/src/charm.py#L189)

Reply:
If this hasn't been fixed yet, lets move on, but it would be worth taking a peak to see if anything has changed since @gruyaume logged the bug.

Reply:
I'm not sure what's the state of this bug. I'd recommend merging as is and opening a new task in the backlog to add authentication to add certificates there. It doesn't make much difference at the end of the day, the metrics are still accessible unauthenticated, it's just that prometheus will check the cert of this endpoint.